core-bundle/contao/templates/_new/component/_table.html.twig

```diff
@@ -124,7 +124,7 @@
                                     <td>
                                         {%- block table_body_cell_content %}
                                             {%- block table_cell_content %}
-                                                {{- cell|nl2br|insert_tag_raw -}}
+                                                {{- cell|raw -}}
```

Comment:
```suggestion
                                                {{- cell|nl2br|raw -}}
```
I think `nl2br` needs to stay to convert `\n` into `<br>\n`

Reply:
I think we cannot use that filter:

> The nl2br filter pre-escapes the input before applying the transformation.

https://twig.symfony.com/doc/2.x/filters/nl2br.html

Reply:
IMHO `nl2br` preescapes things, so this won't work.

Reply:
Also: `|insert_tag` must stay.

Reply:
Added in 894ce41d62a64d0bd09929ba6c080728d689d3a0.

Reply:
Removing `nl2br` is the even bigger BC break IMO, we should still support that I think.
We can probably use something like `|trim|replace({'\n': '<br>\n'})`

Reply:
Because of the [preescaping](https://github.com/twigphp/Twig/blob/609794c37668989b670d79e97baf8d5ce05e0a97/src/Extension/CoreExtension.php#L200) the `nl2br` filter would need to come last, so `cell|…|raw|nl2br`.

But I also noticed, the preescaping messes with our chunked output, so having a preescaping filter in the chain together with `insert_tag_raw` is a bad idea. So I'd agree on using `replace` for this. Maybe, we should even move this to the controller, and make sure no `\n` is in the cell output to begin with. Or would you need to adjust this as a user, wdyt?

Reply:
I think I’d prefer doing it in the template.
```twig
{{- cell|trim|replace({'\n': '<br>\n'})|insert_tag|raw -}}
```

Reply:
`{'\n': '<br>\n'}` won't suffice though, PHPs `nl2br` handles `\r\n`, `\n\r`, `\n` and `\r`.

Reply:
In this case, I think I'm more leaning towards handling this in the controller, though…

Reply:
(I'll also have a look in how we could improve filter preescaping but this is another topic.)

Reply:
In Contao 4 it is handled in the controller. And `nl2br` is not applied, if Contao detects HTML block elements in a table cell:

https://github.com/contao/contao/blob/e581d92fc8d7d2447369acf036265b41ad567852/core-bundle/src/Resources/contao/elements/ContentTable.php#L122-L137

In order to be backwards compatible to Contao 4, the `TableController` in Contao 5 will need to apply the same logic.

Reply:
Ouuhh, I would really like to avoid having this kind of logic in the new code base. Detecting HTML without proper parsing never works out well in the end. I think, if anything, we should apply the transformation depending on a static or per-element config. Any ideas?

Reply:
But the logic needs to be per cell. If you have 100 table cells and you want line breaks in all of them, but only need HTML in 10 of them, you would then have to manually write `<br>` in _all_ table cells.

Reply:
Really? There is a use case where you want a different behavior per cell? I mean, I believe you of course, but that sounds like a lot of magic to me.

Reply:
Can we make the `ContentTable::nl2br()` method available in the template?

Reply:
> I think I’d prefer doing it in the template.
> 
> ```
> {{- cell|trim|replace({'\n': '<br>\n'})|insert_tag|raw -}}
> ```

Added in 13b815438387e5ff4c66149e38d426b0e525e698.

Reply:
> Added in https://github.com/contao/contao/commit/13b815438387e5ff4c66149e38d426b0e525e698.

@leofeyer this isn't fully backwards compatible though, see the previous discussion.

Reply:
> `{'\n': '<br>\n'}` won't suffice though, PHPs `nl2br` handles `\r\n`, `\n\r`, `\n` and `\r`.

`\r\n`, `\n\r`, `\n` should all work with that replace filter, and `\r` alone can never be entered in the backend AFAIK.

Reply:
Yes, but in Contao 4 `nl2br` is only applied, if there are no block HTML elements in the cell.